lib/js_rails_routes/generator.rb

```diff
@@ -57,15 +62,28 @@ def handle_route(route_name, route_path)
       "export function #{route_name}_path(params) { return process('#{route_path}', params, [#{keys.join(',')}]); }"
     end
 
-    def routes
-      @routes ||= Rails.application.routes.routes
-                       .select(&:name)
-                       .map { |r| [r.name, r.path.spec.to_s.split('(')[0]] }
-                       .sort { |a, b| a[0] <=> b[0] }
+    def routes_with_engines
+      engines = Rails::Engine.subclasses.reject { |klass| klass.to_s.downcase =~ exclude_engines }
+
+      engine_routes = engines.map do |engine|
+        routes = make_routes(engine.routes.routes)
+        [engine.to_s, routes]
+      end
+      engine_routes = engine_routes.reject { |_, routes| routes.empty? }
```

Comment:
Please use method chain to avoid reassigning `engine_routes` variable:

```rb
engine_routes = engines
  .map { |engine| [engine.to_s. make_routes(engine.routes.routes)] }
  .reject { |_, routes| routes.empty? }
```

Reply:
Good point.